File: ROS/src/simpleRos/simpleRos/simple_parameter.py
import rclpy
from rcl_interfaces.msg import SetParametersResult
from rclpy.node import Node
from rclpy.parameter import Parameter


class Simple_parameter(Node):
    def __init__(self):
        super().__init__("simple_parameter")
        self.declare_parameter('my_para', '내가 만든 파라미터')
        self.declare_parameter('node_name', 'simple_parameter')
        self.declare_parameter('number1', 123)
        self.declare_parameter('number2', 3.141592)

        self.my_para = self.get_parameter('my_para').get_parameter_value().string_value
        self.node_name = self.get_parameter('node_name').get_parameter_value().string_value
        self.number1 = self.get_parameter('number1').get_parameter_value().integer_value
        self.number2 = self.get_parameter('number2').get_parameter_value().double_value

        self.add_on_set_parameters_callback(self.parameter_callback)

        self.create_timer(1, self.print_parameter)

    def print_parameter(self):
        self.get_logger().info(f"parameter : {self.my_para}")
        self.get_logger().info(f"parameter : {self.node_name}")
        self.get_logger().info(f"parameter : {self.number1}")
        self.get_logger().info(f"parameter : {self.number2}")

    def parameter_callback(self, parameters : list[Parameter]):
        for parameter in parameters:
            if parameter.name == 'my_para':
                # self.my_para = parameter.get_parameter_value().string_value
                self.my_para = parameter.value
            if parameter.name == 'node_name':
                self.node_name = parameter.value
            if parameter.name == 'number1':
                self.number1 = parameter.value
            if parameter.name == 'number2':
                self.number2 = parameter.value
        return SetParametersResult(successful=True)

def main():
    rclpy.init()
    node = Simple_parameter()
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        node.destroy_node()

if __name__ == "__main__":
    main()

File: ROS/src/simple_ros_cpp/src/simple_parameter.cpp
#include "rcl_interfaces/msg/set_parameters_result.hpp"
#include "rclcpp/rclcpp.hpp"

using namespace std;
using namespace std::chrono_literals;

class SimpleParameter : public rclcpp::Node
{
public:
    SimpleParameter()
        : Node("simple_parameter")
    {
        declare_parameter<std::string>("my_para", "내가 만든 파라미터");
        _my_para = get_parameter("my_para").as_string();
        _on_set_para_handle = add_on_set_parameters_callback(std::bind(&SimpleParameter::parameter_callback, this, std::placeholders::_1));
        _print_timer = create_wall_timer(1s, std::bind(&SimpleParameter::print_callback, this));
    }

private:
    std::string _my_para;
    rclcpp::node_interfaces::OnSetParametersCallbackHandle::SharedPtr _on_set_para_handle;
    rclcpp::TimerBase::SharedPtr _print_timer;
    void print_callback()
    {
        RCLCPP_INFO(get_logger(), _my_para.c_str());
    }
    rcl_interfaces::msg::SetParametersResult parameter_callback(const std::vector<rclcpp::Parameter> &parameters)
    {
        for (auto parameter : parameters)
        {
            if (parameter.get_name() == "my_para")
            {
                _my_para = parameter.as_string();
            }
        }
        rcl_interfaces::msg::SetParametersResult result;
        result.successful = true;
        return result;
    }
};

int main()
{
    rclcpp::init(0, nullptr);
    auto node = std::make_shared<SimpleParameter>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}
